vdr/ambassador.go

```diff
@@ -134,16 +144,45 @@ func (n *ambassador) handleCreateDIDDocument(transaction dag.Transaction, propos
 		return err
 	}
 
+	// check for an existing document, it could have been a parallel create. If existing merge and update.
+	currentDIDDocument, currentDIDMeta, err := n.didStore.Resolve(proposedDIDDocument.ID, nil)
+	if err != nil && !errors.Is(err, types.ErrNotFound) {
+		return fmt.Errorf("unable to create DID document: %w", err)
+	}
+	sourceTransactions := []hash.SHA256Hash{transaction.Ref()}
+	var updatedAtP *time.Time
+	if currentDIDDocument != nil {
+		mergedDoc, err := doc.MergeDocuments(*currentDIDDocument, proposedDIDDocument)
+		if err != nil {
+			return fmt.Errorf("unable to merge conflicted DID Document: %w", err)
```

Comment:
```suggestion
			return fmt.Errorf("unable to merge conflicted DID Document (tx=%s): %w", transaction.Ref(), err)
```

Reply:
error wrapping in dag.Publisher adds TX reference